### DISCUSSION TITLE
microstream and client side serialization question

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#514] [Q&A] [answered]
Hi All;
we are currently running on wildfly 23 with JPA and supporting both web clients and Java clients (via open webstart).
supporting Java clients means that we currently send our JPA entities to the client and back (so they are serialized back and forth using regular java serialization). 
considering a microstream porting, I wanted and make sure that there are no issues here (not sure about that). the thing is that if we port, wildfly will use microstream on the backend instread of JPA (using with microstream own serialization mechanism) but the object graph will need to support being sent back to java clients (using the default java serializations becuase I we do not control client installations - they only put a jre there)
would this work?
your assistance is appreciated.

**Comment by fh-ms** [comment, marked as answer]
Hi, 
that is not a problem at all. Just ensure that your entities implement java.io.Serializable, and you're good to go. It doesn't affect MicroStream in any way, since it is just a simple marker interface.
